triton-vm/src/stark.rs

```diff
@@ -424,44 +400,53 @@ impl Stark {
     }
 
     fn compute_quotient_segments(
-        master_main_table: &MasterMainTable,
-        master_aux_table: &MasterAuxTable,
-        fri_domain: ArithmeticDomain,
+        main_table: &mut MasterMainTable,
+        aux_table: &mut MasterAuxTable,
         quotient_domain: ArithmeticDomain,
         challenges: &Challenges,
         quotient_combination_weights: &[XFieldElement],
-    ) -> (Array2<XFieldElement>, Array1<Polynomial<XFieldElement>>) {
-        let calculate_quotients_with_just_in_time_low_degree_extension = || {
+    ) -> (
+        Array2<XFieldElement>,
+        Array1<Polynomial<'static, XFieldElement>>,
+    ) {
+        let (Some(main_quotient_domain_codewords), Some(aux_quotient_domain_codewords)) = (
+            main_table.quotient_domain_table(),
+            aux_table.quotient_domain_table(),
+        ) else {
+            // It is possible that the quotient domain codewords are cached for the main
+            // table, but not for the auxiliary table. To compute the quotients, either both
+            // or neither are needed.[^1] For peak memory consumption, it is beneficial to
+            // clear any unused cache.
+            //
+            // Throwing away any cache here incurs a performance penalty later, when
+            // revealing the rows indicated by FRI. This is deemed an acceptable tradeoff
+            // when peak memory usage is a larger issue than compute time.
+            //
+            // [^1]: Code using exactly one cache _could_ exist, but oh! the engineering.
+            main_table.clear_cache();
+            aux_table.clear_cache();
```

Comment:
The auxiliary table requires less space than the main table. Hence, it is possible that allocation for the quotient domain main table fails with an out-of-memory error, but allocation for the quotient domain auxiliary table succeeds.

Even if the main table required less space than the auxiliary table, it is possible that other programs de-allocate memory during the prover's execution, leading to a cached auxiliary table despite the main table not being cached.

Your remark here highlights that neither scenario is explained in the comment, and I'll amend accordingly. How about:

> The decision to cache the quotient domain main table can be independent of the decision to cache the quotient domain auxiliary table. Additionally, available memory is affected by other running programs. Together, this may result in one table being cached while the other is not. To compute the quotients, […]